examples/pipeline_tacotron2/loss/test_tacotron2_loss.py

```diff
@@ -0,0 +1,197 @@
+import os
+import unittest
+import tempfile
+
+import torch
+from torch.autograd import gradcheck, gradgradcheck
+
+from loss_function import Tacotron2Loss
+
+
+def skipIfNoCuda(test_item):
+    if torch.cuda.is_available():
+        return test_item
+    force_cuda_test = os.environ.get("TORCHAUDIO_TEST_FORCE_CUDA", "0")
+    if force_cuda_test not in ["0", "1"]:
+        raise ValueError('"TORCHAUDIO_TEST_FORCE_CUDA" must be either "0" or "1".')
+    if force_cuda_test == "1":
+        raise RuntimeError(
+            '"TORCHAUDIO_TEST_FORCE_CUDA" is set but CUDA is not available.'
+        )
+    return unittest.skip("CUDA is not available.")(test_item)
+
+
+class TempDirMixin:
+    """Mixin to provide easy access to temp dir"""
+
+    temp_dir_ = None
+
+    @classmethod
+    def get_base_temp_dir(cls):
+        # If TORCHAUDIO_TEST_TEMP_DIR is set, use it instead of temporary directory.
+        # this is handy for debugging.
+        key = "TORCHAUDIO_TEST_TEMP_DIR"
+        if key in os.environ:
+            return os.environ[key]
+        if cls.temp_dir_ is None:
+            cls.temp_dir_ = tempfile.TemporaryDirectory()
+        return cls.temp_dir_.name
+
+    @classmethod
+    def tearDownClass(cls):
+        super().tearDownClass()
+        if cls.temp_dir_ is not None:
+            cls.temp_dir_.cleanup()
+            cls.temp_dir_ = None
+
+    def get_temp_path(self, *paths):
+        temp_dir = os.path.join(self.get_base_temp_dir(), self.id())
+        path = os.path.join(temp_dir, *paths)
+        os.makedirs(os.path.dirname(path), exist_ok=True)
+        return path
+
+
+def _get_inputs(dtype, device):
+    n_mel, n_batch, max_mel_specgram_length = 3, 2, 4
+    mel_specgram = torch.rand(
+        n_batch, n_mel, max_mel_specgram_length, dtype=dtype, device=device
+    )
+    mel_specgram_postnet = torch.rand(
+        n_batch, n_mel, max_mel_specgram_length, dtype=dtype, device=device
+    )
+    gate_out = torch.rand(n_batch, dtype=dtype, device=device)
+    truth_mel_specgram = torch.rand(
+        n_batch, n_mel, max_mel_specgram_length, dtype=dtype, device=device
+    )
+    truth_gate_out = torch.rand(n_batch, dtype=dtype, device=device)
+
+    return (
+        mel_specgram,
+        mel_specgram_postnet,
+        gate_out,
+        truth_mel_specgram,
+        truth_gate_out,
+    )
+
+
+class Tacotron2LossTest(unittest.TestCase, TempDirMixin):
```

Comment:
please follow the convention we have for testing cpu/gpu

Reply:
please also add a test that validates the shape of the output.

Reply:
Thanks for the suggestion.
The shape test is [here](https://github.com/pytorch/audio/pull/1625/files#diff-2405e8a4b292ae3cadb917a86765e3158d22489dc14f00e895c25bdf19604e22R63).
And the base class and subclasses is also addressed in the same [commit](https://github.com/pytorch/audio/pull/1625/commits/5b9942bc3f07e4b314a2a5e25a6c2b172379bce4).